src/bake/DevServer.zig

```diff
@@ -621,29 +795,162 @@ fn theRealBundlingFunction(
+        payload.appendAssumeCapacity(MessageId.errors.char());
+        const w = payload.writer();
+
+        try w.writeInt(u32, @intCast(dev.incremental_result.failures_removed.items.len), .little);
+
+        for (dev.incremental_result.failures_removed.items) |removed| {
+            try w.writeInt(u32, @bitCast(removed.getOwner().encode()), .little);
+            removed.deinit();
+        }
+
+        for (dev.incremental_result.failures_added.items) |added| {
+            try w.writeAll(added.data);
+
+            switch (added.getOwner()) {
+                .none, .route => unreachable,
+                .server => |index| try dev.server_graph.traceDependencies(index, .no_stop),
+                .client => |index| try dev.client_graph.traceDependencies(index, .no_stop),
+            }
+        }
+
+        for (dev.incremental_result.routes_affected.items) |route_index| {
+            const route = &dev.routes[route_index.get()];
+            route.server_state = .possible_bundling_failures;
+        }
+
+        _ = dev.app.publish(DevWebSocket.global_channel, payload.items, .binary, false);
+    } else if (dev.incremental_result.failures_removed.items.len > 0) {
+        if (dev.bundling_failures.count() == 0) {
+            _ = dev.app.publish(DevWebSocket.global_channel, &.{MessageId.errors_cleared.char()}, .binary, false);
+            for (dev.incremental_result.failures_removed.items) |removed| {
+                removed.deinit();
+            }
+        } else {
+            var payload = try std.ArrayList(u8).initCapacity(sfa, @sizeOf(MessageId) + @sizeOf(u32) + dev.incremental_result.failures_removed.items.len * @sizeOf(u32));
+            defer payload.deinit();
+            payload.appendAssumeCapacity(MessageId.errors.char());
+            const w = payload.writer();
+
+            try w.writeInt(u32, @intCast(dev.incremental_result.failures_removed.items.len), .little);
+
+            for (dev.incremental_result.failures_removed.items) |removed| {
+                try w.writeInt(u32, @bitCast(removed.getOwner().encode()), .little);
+                removed.deinit();
+            }
+
+            _ = dev.app.publish(DevWebSocket.global_channel, payload.items, .binary, false);
+        }
+    }
+
+    dev.incremental_result.failures_removed.clearRetainingCapacity();
+}
+
+/// Used to generate the entry point. Unlike incremental patches, this always
+/// contains all needed files for a route.
+fn generateClientBundle(dev: *DevServer, route: *Route) bun.OOM![]const u8 {
+    assert(route.client_bundle == null);
+    assert(route.server_state == .loaded); // page is unfit to load
+
+    dev.graph_safety_lock.lock();
+    defer dev.graph_safety_lock.unlock();
+
+    // Prepare bitsets
+    var sfa_state = std.heap.stackFallback(65536, dev.allocator);
+
+    const sfa = sfa_state.get();
+    dev.server_graph.affected_by_trace = try DynamicBitSetUnmanaged.initEmpty(sfa, dev.server_graph.bundled_files.count());
```

Comment:
Is this going to overwrite data in `affected_by_tree`? If not, the memory isn't reused, could we just remove `affected_by_tree` from the incremental graph struct?

Reply:
the memory is not reused. i honestly dont want to do this refactor right now, but it should just be an argument passed to traceDependencies/traceImports